src/connector/src/source/filesystem/s3/enumerator.rs

```diff
@@ -99,15 +99,14 @@ impl SplitEnumerator for S3SplitEnumerator {
     }
 
     async fn list_splits(&mut self) -> crate::error::ConnectorResult<Vec<Self::Split>> {
-        let mut objects = Vec::new();
-        loop {
-            let (files, has_finished) = self.get_next_page::<FsSplit>().await?;
-            objects.extend(files);
-            if has_finished {
-                break;
-            }
-        }
-        Ok(objects)
+        // fetch one page as validation, no need to get all pages
+        let (_, _) = self.get_next_page::<FsSplit>().await?;
+
+        Ok(vec![FsSplit {
+            name: "empty_split".to_owned(),
+            offset: 0,
+            size: 0,
+        }])
     }
```

Comment:
Is this correct for legacy s3 enumerator? IIUC its splits need to be enumerated in meta 🤔